src/libArgParse/Alternation.hpp

```diff
@@ -102,6 +102,16 @@ class Alternation : public GrammarElement
                         //std::cout << "  Alternation"<< std::to_string(m_instanceId) << ": have possible candidate: '" << candidate->getMatchedString() << "'" << std::endl;
                         candidateList.push_back(candidate);
                     }
+
+                    if(childRc.errorType == ParseRc::ErrorType::retrievingGrammarFailed)
+                    {
+                        rc.errorType = ParseRc::ErrorType::retrievingGrammarFailed;
+                    }
+                }
+
+                if(childRc.isBad())
+                {
+                    rc.ErrorMessage = childRc.ErrorMessage;
```

Comment:
only do this for retrievingGrammarFaile cases.
As we expect most of the alternation choices to fail during parse. We are then only interested in errors from grammar retrieval.